src/recovery.rs

```diff
@@ -514,13 +521,15 @@ impl Recovery {
         false
     }
 
+    // TODO: move to Congestion Control and implement draft 24
     fn in_persistent_congestion(&mut self, _largest_lost_pkt: &Sent) -> bool {
         let _congestion_period = self.pto() * PERSISTENT_CONGESTION_THRESHOLD;
 
         // TODO: properly detect persistent congestion
         false
     }
 
+    // TODO: move to Congestion Control
```

Comment:
Should we though? Each CC implementation is going to have its own `CongestionEvent()` method, but I think the rest would stay the same?

Reply:
OnPacketLost() is mentioned in Congestion Control pseudocode in the draft, so try to follow it first. In reality as you say it looks good enough to keep OnPacketsLost() in Recovery and CC need to implement CongestionEvent() and InPersistentCongestion(). However set_bytes_in_flight() is called here too, so probably a good idea to move into CC, but there are many other fields accessed from Recovery (such as lost_count, sent[], etc).

So, for me it's hard to decide to move it or not... to keep the changes minimum I only left the TODO comment. This is a typical real implementation vs pseudocode issue. I don't want to push but basically a reminder to think it again if there is a good way to move into CC.